### PR TITLE
Postgres Source: add timezone awareness and handle BC dates

### LINKED ISSUE #8903: Postgres source date includes time, and cannot handle BC dates
We can make Postgres source return the correct date only format. However, the change is reverted ([commit](https://github.com/airbytehq/airbyte/pull/8726/commits/e1dac8ed19197f7e66235a9f7f96967fb6c8db41)) in #8726, because we are not sure if the destination / dbt can handle a BC date string correctly.

### LINKED ISSUE #12905: 🎉 Source Postgres: Describe timezone awareness
## Tell us about the problem you're trying to solve
As part of the [[EPIC] Expand date-time schema declarations to describe timezone awareness](https://github.com/airbytehq/airbyte/issues/8280) we would like to add to the Source Postgress additional information to the JSON schema about timestamp which is timezone aware/unaware.

There are next date-related types that should be covered:
- timestamp with/without timezone
- date
- time with/without timezone

## Describe the solution you’d like

- [ ] Add `airbyte_type_format` information on top of JSON Schema types This field can be used to say that a `string, format: date-time` is `airbyte_type_format: timestamp_with_timezone` or `without_timezone`. 
- [ ] make sure that during date conversion we do not include 'Z' for date-time data type
- [ ] source acceptance tests should verify that sources output the correct types

## Additional context
It is a common practice for all DB's that for all time-related data types except timestamp, it doesn't care about TZ and returns a constant time value irrespective of the DB timezone.
For example, we have date-time with value "2011-10-04 12:58:36", so with session timezone +00:00 or -06:00 it will return the same value "2011-10-04 12:58:36

### PR DESCRIPTION
**Pull request by VitaliiMaltsev**
## What

1. Cant handle any BC dates (Ex. '199-10-10 BC" would be missed BC attribute and returned as an AD date)
2. As part of the [[EPIC] Expand date-time schema declarations to describe timezone awareness](https://github.com/airbytehq/airbyte/issues/8280) we would like to add to the Source Postgres additional information to the JSON schema about timestamp which is timezone aware/unaware.

There are next date-related types that should be covered:

timestamp with/without timezone
date
time with/without timezone

## How
1. When converting an object from resultset metadata to LocalDate class, we can use the method getEra() 
2. Created new JsonSchemaType final variable to cover all of the Postgres date-time types:
Screenshot 2022-05-26 at 17 55 50

## Recommended reading order

1. `JsonSchemaType.java`
2. `AbstractJdbcCompatibleSourceOperations.java`
3. `PostgresSourceOperations.java`

## 🚨 User Impact 🚨
user impact is possible at the normalization stage, since normalization cannot yet handle new parameters in the json schema

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/SUMMARY.md`
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

**Review by alexandr-shegeda — APPROVED** [review]
LGTM

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 1]
```diff
@@ -80,8 +95,12 @@ public void setJsonField(final ResultSet resultSet, final int colIndex, final Ob
...
     } else if (columnTypeName.equalsIgnoreCase("bytea")) {
       putString(json, columnName, resultSet, colIndex);
-    } else if (columnTypeName.equalsIgnoreCase("time") || columnTypeName.equalsIgnoreCase("timetz")) {
-      putString(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("time")) {
+      putTime(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("timetz")) {
+      putTimeWithTimezone(json, columnName, resultSet, colIndex);
+    } else if (columnTypeName.equalsIgnoreCase("timestamptz")) {
+      putTimestampWithTimezone(json, columnName, resultSet, colIndex);
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java:103]: Is it possible we will have more `else if` statements here, in the future?
I suggest refactoring this block to smth like:
```java
 return switch (typeName) {
      case "real", "double", "float" -> Double.parseDouble(varCharValue);
      case "varchar" -> varCharValue;
      case "boolean" -> Boolean.parseBoolean(varCharValue);
      case "integer" -> Integer.parseInt(varCharValue);
      default -> null;
    };
```

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java [hunk 2]
```diff
@@ -246,4 +250,27 @@ public String getFullyQualifiedTableNameWithQuoting(final Connection connection,
...
+    OffsetDateTime timestamptz = getDateTimeObject(resultSet, index, OffsetDateTime.class);
+    LocalDate localDate = timestamptz.toLocalDate();
+    node.put(columnName, resolveEra(localDate, timestamptz.toString()));
+  }
+
+  protected String resolveEra(LocalDate date, String value) {
+    return isBCE(date) ? value.substring(1) + " BC" : value;
+  }
+
+  private static boolean isBCE(LocalDate date) {
```
Comment (alexandertsukanov) [inline airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java:272]: why `static` ?

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/java/io/airbyte/integrations/source/postgres/PostgresStrictEncryptJdbcSourceAcceptanceTest.java [hunk 3]
```diff
@@ -7,17 +7,25 @@
...
 import io.airbyte.commons.string.Strings;
+import io.airbyte.db.jdbc.JdbcSourceOperations;
 import io.airbyte.integrations.base.Source;
 import io.airbyte.integrations.base.ssh.SshHelpers;
 import io.airbyte.integrations.source.jdbc.JdbcSource;
 import io.airbyte.integrations.source.jdbc.test.JdbcSourceAcceptanceTest;
-import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.integrations.source.relationaldb.models.DbStreamState;
+import io.airbyte.protocol.models.*;
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/java/io/airbyte/integrations/source/postgres/PostgresStrictEncryptJdbcSourceAcceptanceTest.java:24]: Please, don't use wildcard imports.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix&nbsp; | &nbsp;**Tags:** Brittle

File: airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java (hunk 2 above)
Comment (VitaliiMaltsev) [reply airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java:272]: it is util method and we can reuse it without initialising of the new instance AbstractJdbcCompatibleSourceOperations

File: airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresJdbcSourceAcceptanceTest.java [hunk 4]
```diff
@@ -7,16 +7,23 @@
...
 import io.airbyte.commons.json.Jsons;
 import io.airbyte.commons.resources.MoreResources;
 import io.airbyte.commons.string.Strings;
+import io.airbyte.db.jdbc.JdbcSourceOperations;
 import io.airbyte.integrations.source.jdbc.AbstractJdbcSource;
 import io.airbyte.integrations.source.jdbc.test.JdbcSourceAcceptanceTest;
-import io.airbyte.protocol.models.ConnectorSpecification;
+import io.airbyte.integrations.source.relationaldb.models.DbState;
+import io.airbyte.integrations.source.relationaldb.models.DbStreamState;
+import io.airbyte.protocol.models.*;
```
Comment (alexandertsukanov) [inline airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresJdbcSourceAcceptanceTest.java:22]: Please, don't use wildcard imports.

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :hammer_and_wrench: This code needs a fix

File: airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java (hunk 2 above)
Comment (alexandertsukanov) [reply airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java:272]: @VitaliiMaltsev in such case, should it be `public`?

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :question: I have a question

**Review by alexandertsukanov — CHANGES_REQUESTED** [review]
Finished iteration of review. Left some comments

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Tags:** Reusable

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java (hunk 1 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java]: > Is it possible we will have more `else if` statements here, in the future? I suggest refactoring this block to smth like:
> 
> ```java
>  return switch (typeName) {
>       case "real", "double", "float" -> Double.parseDouble(varCharValue);
>       case "varchar" -> varCharValue;
>       case "boolean" -> Boolean.parseBoolean(varCharValue);
>       case "integer" -> Integer.parseInt(varCharValue);
>       default -> null;
>     };
> ```
> 
> __ [sema-logo](https://semasoftware.com/gh) [ sema-logo ](https://semasoftware.com/gh) [ ](https://semasoftware.com/gh)  **Summary:** 🛠️ This code needs a fix

replaced with switch

File: airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java (hunk 2 above)
Comment (VitaliiMaltsev) [reply airbyte-db/lib/src/main/java/io/airbyte/db/jdbc/AbstractJdbcCompatibleSourceOperations.java]: > @VitaliiMaltsev in such case, should it be `public`?
> 
> __ [sema-logo](https://semasoftware.com/gh) [ sema-logo ](https://semasoftware.com/gh) [ ](https://semasoftware.com/gh)  **Summary:** ❓ I have a question

made it public

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/java/io/airbyte/integrations/source/postgres/PostgresStrictEncryptJdbcSourceAcceptanceTest.java (hunk 3 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/java/io/airbyte/integrations/source/postgres/PostgresStrictEncryptJdbcSourceAcceptanceTest.java]: > Please, don't use wildcard imports.
> 
> __ [sema-logo](https://semasoftware.com/gh) [ sema-logo ](https://semasoftware.com/gh) [ ](https://semasoftware.com/gh)  **Summary:** 🛠️ This code needs a fix  |  **Tags:** Brittle

removed star import

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 5]
```diff
@@ -116,6 +154,10 @@ public JDBCType getFieldType(final JsonNode field) {
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
         return JDBCType.VARCHAR;
+      } else if (typeName.equalsIgnoreCase("timestamptz")) {
+        return JDBCType.TIMESTAMP_WITH_TIMEZONE;
+      } else if (typeName.equalsIgnoreCase("timetz")) {
+        return JDBCType.TIME_WITH_TIMEZONE;
       }
```
[the remark this answers is not in the document]
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java]: > Looks like, the comment above is related to this block of code as well.

replaced with switch

File: airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresJdbcSourceAcceptanceTest.java (hunk 4 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres/src/test/java/io/airbyte/integrations/source/postgres/PostgresJdbcSourceAcceptanceTest.java]: > Please, don't use wildcard imports.
> 
> __ [sema-logo](https://semasoftware.com/gh) [ sema-logo ](https://semasoftware.com/gh) [ ](https://semasoftware.com/gh)  **Summary:** 🛠️ This code needs a fix

removed star import

**Review by alexandertsukanov — APPROVED** [review]
LGTM

__
[sema-logo](https://semasoftware.com/gh) &nbsp;**Summary:** :ok_hand: This code looks good

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/resources/expected_spec.json [hunk 6]
```diff
@@ -54,11 +54,17 @@
         "airbyte_secret": true,
         "order": 5
       },
+      "jdbc_url_params": {
```
Comment (grishick) [inline airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/resources/expected_spec.json:57]: I don't think this change in expected_spec.json is related to this PR. How did it end up here?

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 7]
```diff
@@ -67,58 +72,87 @@ public JsonNode rowToJson(final ResultSet queryContext) throws SQLException {
...
-        // BIT is a bit string in Postgres, e.g. '0100'
-        case BIT, CHAR, VARCHAR, LONGVARCHAR -> putString(json, columnName, resultSet, colIndex);
-        case DATE -> putDate(json, columnName, resultSet, colIndex);
-        case TIME -> putTime(json, columnName, resultSet, colIndex);
-        case TIMESTAMP -> putTimestamp(json, columnName, resultSet, colIndex);
-        case BLOB, BINARY, VARBINARY, LONGVARBINARY -> putBinary(json, columnName, resultSet, colIndex);
-        case ARRAY -> putArray(json, columnName, resultSet, colIndex);
-        default -> putDefault(json, columnName, resultSet, colIndex);
+      switch (columnTypeName) {
+        case "bool", "boolean" -> putBoolean(json, columnName, resultSet, colIndex);
```
Comment (grishick) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java:95]: why do we need these extra `case` statements for `bool`, `boolean`, and `bytea` in the context of this PR?

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 8]
```diff
@@ -67,58 +72,87 @@ public JsonNode rowToJson(final ResultSet queryContext) throws SQLException {
...
+        case "bool", "boolean" -> JDBCType.BOOLEAN;
         // BYTEA is variable length binary string with hex output format by default (e.g. "\x6b707a").
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
-        return JDBCType.VARCHAR;
-      }
-
-      return JDBCType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt());
+        case "bytea" -> JDBCType.VARCHAR;
+        case "timestamptz" -> JDBCType.TIMESTAMP_WITH_TIMEZONE;
```
Comment (grishick) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java:152]: Strings `timestamptz` and `timetz` are used more than once in the codebase - please create `public static` constants for them.

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java [hunk 9]
```diff
@@ -67,58 +72,87 @@ public JsonNode rowToJson(final ResultSet queryContext) throws SQLException {
...
+      return switch (typeName) {
+        case "bool", "boolean" -> JDBCType.BOOLEAN;
         // BYTEA is variable length binary string with hex output format by default (e.g. "\x6b707a").
         // It should not be converted to base64 binary string. So it is represented as JDBC VARCHAR.
         // https://www.postgresql.org/docs/14/datatype-binary.html
-        return JDBCType.VARCHAR;
-      }
-
-      return JDBCType.valueOf(field.get(INTERNAL_COLUMN_TYPE).asInt());
+        case "bytea" -> JDBCType.VARCHAR;
```
Comment (grishick) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java:151]: same question as above - how is `bytea` relevant to TZ awareness and BC date?

**Review by grishick — CHANGES_REQUESTED** [review]
Looks great overall. I added a couple of minor comments

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/resources/expected_spec.json (hunk 6 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres-strict-encrypt/src/test/resources/expected_spec.json:57]: > I don't think this change in expected_spec.json is related to this PR. How did it end up here?

Postgres Source Strict encrypt tests were broken in master branch, so i decided to fix them in scope of this PR

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java (hunk 8 above)
Comment (VitaliiMaltsev) [reply airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/PostgresSourceOperations.java]: > Strings `timestamptz` and `timetz` are used more than once in the codebase - please create `public static` constants for them.

done

**Review by grishick — APPROVED** [review]

File: airbyte-integrations/connectors/source-postgres-strict-encrypt/Dockerfile [hunk 10]
```diff
@@ -16,5 +16,5 @@ ENV APPLICATION source-postgres-strict-encrypt
 
 COPY --from=build /airbyte /airbyte
 
-LABEL io.airbyte.version=0.4.15
+LABEL io.airbyte.version=0.4.16
```
Comment (alexandr-shegeda) [inline airbyte-integrations/connectors/source-postgres-strict-encrypt/Dockerfile:19]: please bump the version, the latest actual in docker hub is [0.4.18](https://hub.docker.com/layers/source-postgres/airbyte/source-postgres/0.4.18/images/sha256-de3f1cd87721f399d808b998791a9477220d521b7986b50ba386d79013942600?context=explore)

**Comment by VitaliiMaltsev** [comment]
@grishick 
I can't publish this connector because CI tests for some reason can't run postgres test containers.
```
   2022-05-30 13:16:33 INFO o.t.c.GenericContainer(tryStart):372 - Creating container for image: postgres:13-alpine
    2022-05-30 13:16:33 ERROR o.t.c.GenericContainer(tryStart):524 - Could not start container
    java.lang.RuntimeException: java.io.IOException: Broken pipe
```
Locally, these tests work well and run the test containers

**Comment by grishick** [comment]
/publish connector=connectors/source-postgres run-tests=false

**Comment by VitaliiMaltsev** [comment]
/publish connector=connectors/source-postgres-strict-encrypt run-tests=false

**Comment by grishick** [comment]
@evantahler FYI auto-bump has failed here (see @VitaliiMaltsev 's last attempt)

**Comment by evantahler** [comment]
@grishick isn't a failure expected if the version on this branch has already been bumped (your previous /publish did work).   The previous failures of the publish command seem to be test failures, as noted above

```
SshPasswordPostgresSourceAcceptanceTest > testDiscover() FAILED
    org.testcontainers.containers.ContainerLaunchException: Container startup failed
        at app//org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:345)
        at app//org.testcontainers.containers.GenericContainer.start(GenericContainer.java:326)
(snip...)
```

**Comment by grishick** [comment]
> Seems to imply that if we are bumping the version, in all cases, we want to update our `*definitions.yaml`. But, these strict-encrypt versions aren't listed there, so this command will fail & exit 1.

Hm... so this means that publishing strict-encrypt connectors with `publish` command is broken, but we still need to be publishing them to docker hub unless we replace these connectors entirely with some other way of enforcing encryption in the cloud. I am not aware of any work to get rid of strict encrypt connectors though.

**Comment by tuliren** [comment]
The publication of the `strict-encrypt` version succeeded. It was the version bumping that failed. This is expected, because the `strict-encrypt` version only exists on cloud.

The current workflow is:
1. Publish the normal version and the `strict-encrypt` version.
2. Ignore the version bumping failure from the `strict-encrypt` version.
3. Bump the `strict-encrypt` version in `airbyte-cloud`.

I guess we can just make the version bumping script ignore the error if a connector does not exist.

> I also think that we might be moving away from these secondary image and use an ENV var / setting to toggle on strict-encrypt mode in cloud. @tuliren - what's the path forward here?

Yes, I think we can do the following:
1. Add a cloud environment variable that is specific to the Airbyte cloud deployment.
    - Or reuse an existing one if it exists. 
2. Pass the environment variable to the connector container.
3. When the connector detects such an environment variable, it updates the spec and behave just like the `strict-encrypt` version.

Added a comment to this related issue: https://github.com/airbytehq/airbyte/issues/8634#issuecomment-1144658962

**Comment by VitaliiMaltsev** [comment]
> After updating our Postgres source connector to 0.4.19 following this PR merge, we're witnessing issues in one of our pipelines because date field precision is now truncated if the value of the millisecond or second is 0.
> 
> Value in PG database: `2022-07-02T16:09:00.000000Z` Value exported to BigQuery with Postgres source 0.4.18: `2022-07-02T16:09:00.000000Z` Value exported to BigQuery with Postgres source 0.4.18: `2022-07-02T16:09`
> 
> I'm not sure if it's on purpose or if it's a bug, we've had to revert to 0.4.18 though
@quantson what datatype are you using in postgres for this value? (2022-07-02T16:09:00.000000Z)

**Comment by VitaliiMaltsev** [comment]
@quantson please take a look https://github.com/airbytehq/airbyte/pull/13549

**Pull request merged**